packages/web/components/bridge/immersive/immersive-bridge.tsx

```diff
@@ -0,0 +1,257 @@
+            enter="transition-opacity duration-300"
+            enterFrom="opacity-0"
+            enterTo="opacity-100"
+            leave="transition-opacity duration-150"
+            leaveFrom="opacity-100"
+            leaveTo="opacity-0"
+            afterLeave={() => {
+              setSelectedAssetDenom(undefined);
+              setStep(ImmersiveBridgeScreens.Asset);
+            }}
+          >
+            <ModalCloseButton onClick={() => onClose()} />
+            {step !== ImmersiveBridgeScreens.Asset && (
+              <IconButton
+                onClick={() => {
+                  const previousStep = Number(step) - 1;
+                  // @ts-expect-error
+                  setStep(previousStep);
+                }}
+                className={
+                  "absolute left-8 top-[28px] z-50 w-fit text-osmoverse-400 hover:text-osmoverse-100"
+                }
+                icon={<Icon id="chevron-left" width={16} height={16} />}
+                aria-label="Go Back"
+              />
+            )}
+
+            <div className="flex h-full w-full max-w-[30rem] flex-col gap-10 py-12">
+              <StepProgress
+                className="w-full"
+                steps={[
+                  {
+                    displayLabel: "Asset",
+                    onClick:
+                      step !== ImmersiveBridgeScreens.Asset
+                        ? () => setStep(ImmersiveBridgeScreens.Asset)
+                        : undefined,
+                  },
+                  {
+                    displayLabel: "Amount",
+                    onClick:
+                      step === ImmersiveBridgeScreens.Review
+                        ? () => setStep(ImmersiveBridgeScreens.Amount)
+                        : undefined,
+                  },
+                  {
+                    displayLabel: "Review",
+                  },
+                ]}
+                currentStep={Number(step)}
+              />
+
+              <div className="flex-1">
+                <Screen screenName={ImmersiveBridgeScreens.Asset}>
+                  {({ setCurrentScreen }) => (
+                    <AssetSelectScreen
+                      type={direction}
+                      onSelectAsset={(asset) => {
+                        setCurrentScreen(ImmersiveBridgeScreens.Amount);
+                        setSelectedAssetDenom(asset.coinDenom);
+                      }}
+                    />
+                  )}
+                </Screen>
+                <Screen screenName={ImmersiveBridgeScreens.Amount}>
+                  {() => (
+                    <AmountScreen
+                      type={direction}
+                      assetsInOsmosis={canonicalAssetsWithVariants}
+                    />
+                  )}
+                </Screen>
+                <Screen screenName={ImmersiveBridgeScreens.Review}>
+                  {({ goBack }) => (
+                    <div>
+                      <h6>Step 3: Review</h6>
```

Comment:
Also, localized

Reply:
This is a temp value so I think we can skip it